onnx/shape_inference/implementation.cc

```diff
@@ -660,7 +703,7 @@ std::vector<const TypeProto*> GraphInferencerImpl::doInferencing(
         continue;
       }
     }
-
+    materializeSymbolicShape(inferredInput, symbolTable);
```

Comment:
I think the case of GraphInferencer we should first merge and then cal materializeSymbolicShape on graphInput instead of inferredInput...

Reply:
The original expected flow is TypeandshapeInferencefunction -> generated symbols for unknown types -> attempt Partial DataPropagation -> merge.

Are you saying that moving generated symbols and partial DataPropagation after merge since they can manipulate  non-const graphInput after merge?

I have a concern: do we need `checkTensorShapesAndTypes` for checking the shape after Partial DataPropagation? Currently shape_inference does checkTensorShapesAndTypes in merge function. If we move merge before generated symbols and partial DataPropagation, it won't check the final propagated shape with existing shape.

Reply:
After offline discussion, I agreed with you. Since the inferredInput here isn't inferred in this level, it has generated symbol previously and it should be OK for us to do merge first then materializeSymbolicShape. In that case, we also don't need to use const_cast to break the const... Updated. Thanks!